### PR TITLE
PIN-1372 purpose-process GET /purposes/{purposeId}/versions/{versionId}/documents/{documentId}

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiMarshallerImpl.scala [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+import scala.io.{BufferedSource, Codec}
 
 object PurposesApiMarshallerImpl extends PurposesApiMarshaller with SprayJsonSupport with DefaultJsonProtocol {
 
   override implicit def toEntityMarshallerProblem: ToEntityMarshaller[Problem] = entityMarshallerProblem
 
   override implicit def toEntityMarshallerPurposes: ToEntityMarshaller[Purposes] = sprayJsonMarshaller[Purposes]
+
+  override implicit def toEntityMarshallerFile: ToEntityMarshaller[File] =
+    Marshaller.withFixedContentType(ContentTypes.`application/octet-stream`) { f =>
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiMarshallerImpl.scala]: we could centralize this marshaller like the `entityMarshallerProblem`, as it is/will be used in several services

File: src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala [hunk 2]
```diff
@@ -72,4 +72,15 @@ object BFFErrors {
...
       extends ComponentError("0016", s"No descriptor found in eService $eServiceId")
 
+  final case class InvalidRiskAnalysisContentType(
+    contentType: String,
+    purposeId: String,
+    versionId: String,
+    documentId: String,
+    errors: List[ErrorInfo]
+  ) extends ComponentError(
+        "0016",
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/error/BFFErrors.scala]:
```suggestion
        "0017",
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/error/Handlers.scala [hunk 3]
```diff
@@ -47,6 +49,7 @@ object Handlers {
     case Failure(err: AttributeNotExists)                       => internalServerError(err, logMessage)
     case Failure(err: UnknownTenantOrigin)                      => badRequest(err, logMessage)
     case Failure(err: InvalidInterfaceFileDetected)             => badRequest(err, logMessage)
+    case Failure(err: InvalidRiskAnalysisContentType)           => badRequest(err, logMessage)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/error/Handlers.scala]: we can remove this, it is considered a 500 status in this case

File: src/test/resources/application-test.conf [hunk 4]
```diff
@@ -34,10 +34,14 @@ backend-for-frontend {
...
+          path = "risk-analysis/docs"
+    }
       consumer-documents {
           container = "container"
           path = "agreement/consumer-docs"
-      }
+    }
        eservice-documents {
         container = "container"
         path = "eservices/docs"
```
Comment (galales) [inline src/test/resources/application-test.conf]: indentation

**Review by galales — APPROVED** [review]
🚀

**Review by TonioGela — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**